### PR TITLE
CONTRIBUTING.md - initial commit

### PR DESCRIPTION
**Pull request by Karm**
A stub file.

### CODE REVIEW

**Comment by m-czernek** [comment]
@Karm  I have provided a basic contribution document. WDYT?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+1. Adhere to the review provided within the pull request. If you disagree, we are always open to a debate.
+1. Adhere to the Apache software Foundation's [Code of Conduct](http://www.apache.org/foundation/policies/conduct.html).
+1. Provide integration and unit test where applicable.
+1. Provide full documentation for your changes where applicable.
+1. Provide reasonable commit messages.
+1. Adhere to our codestyle. 
+
+We use the same coding style as [Wildfly](https://github.com/wildfly/wildfly-core/tree/master/ide-configs) with the following exceptions:
+ - Indentation size is 2 spaces.
+ - Contined indentation is 4 spaces.
```
Comment (Karm) [inline CONTRIBUTING.md]: @m-czernek, 
Hmm, ad Indentation/Contined indentation -- do we though? I don't want us to be influenced by the legacy Groovy style. I would like us to follow Sunstone in the style and possibly also release config etc.  They use Wildfly style without modifications IIUC: [see the pom](https://github.com/wildfly-extras/sunstone/blob/master/pom.xml#L60).

Comment (m-czernek) [reply CONTRIBUTING.md]: We definitely do follow 2 spaces tab - 4 spaces continuation at this moment...  Do we want to re-format Noe Core? Seems like [Wildfly](https://github.com/wildfly/wildfly-core/blob/master/ide-configs/eclipse/wildfly-formatter.xml#L43) indeed is different in the continued indentation.

**Comment by m-czernek** [comment]
@Karm could we merge this? Imho the point about indentation is moot, since it's in the README.md. If you want, we can have a further discussion about whether to reformat or not, but for the time being, wf formatter it is.

**Pull request merged**